internal/confmapprovider/discovery/discoverer.go

```diff
@@ -117,6 +119,38 @@ func newDiscoverer(logger *zap.Logger) (*discoverer, error) {
 	return d, nil
 }
 
+func (d *discoverer) discoveryAuthConfFromEnv(discoveryReceiverRaw map[string]any) (*confmap.Conf, error) {
+	yamlProvider := yamlprovider.New()
+	envProvider := envprovider.New()
```

Comment:
I believe these can be globals rather than created each call.

Reply:
got it.